features/nfc/nfc/NFCController.h

```diff
@@ -0,0 +1,183 @@
+             * A remote initiator was discovered (the local controller is in target mode).
+             * @param[in] the NFCRemoteInitiator instance
+             */
+            virtual void on_nfc_initiator_discovered(const SharedPtr<NFCRemoteInitiator>& nfc_initiator) {}
+
+            /**
+             * A remote target was discovered (the local controller is in initiator mode).
+             * @param[in] the NFCRemoteTarget instance
+             */
+            virtual void on_nfc_target_discovered(const SharedPtr<NFCRemoteTarget>& nfc_target) {}
+        };
+
+        /** 
+         * Construct a NFCController instance.
+         *  
+         * @param[in] driver a pointer to a NFCControllerDriver instance
+         * @param[in] queue a pointer to the events queue to use
+         * @param[in] ndef_buffer a bytes array used to store NDEF messages
+         * @param[in] ndef_buffer_sz the array size in bytes
+         */
+        NFCController(NFCControllerDriver* driver, events::EventQueue* queue, uint8_t* ndef_buffer, size_t ndef_buffer_sz);
+
+        /** 
+         * Initialize the NFC controller
+         * 
+         * This method must be called before any other method call.
+         *
+         * @return NFC_OK, or an error.
+         */
+        nfc_err_t initialize();
+
+        /** 
+         * Set the delegate that will receive events generated by this controller.
+         * 
+         * @oaram[in] delegate the delegate instance to use
+         */ 
+        void set_delegate(Delegate* delegate);
+
+        /**
+         * Get the list of RF protocols supported by this controller.
+         * 
+         * @return a bitmask of RF protocols supported by the controller
+         */
+        nfc_rf_protocols_bitmask_t get_supported_rf_protocols() const;
+
+        /**
+         * Set the list of RF protocols to look for during discovery.
+         * 
+         * @param[in] rf_protocols the relevant bitmask
+         * @return NFC_OK on success, or 
+         *  NFC_ERR_UNSUPPORTED if a protocol is not supported by the controller, 
+         *  NFC_ERR_BUSY if the discovery process is already running
+         */ 
+        nfc_err_t configure_rf_protocols(nfc_rf_protocols_bitmask_t rf_protocols);
+
+        /**
+         * Start the discovery process using the protocols configured previously.
+         * 
+         * If remote endpoints are connected when this is called, they will be disconnected.
+         * 
+         * @return NFC_OK on success, or 
+         *  NFC_ERR_BUSY if the discovery process is already running
+         */ 
+        nfc_err_t start_discovery();
+
+        /**
+         * Cancel/stop a running discovery process.
+         * 
+         * @return NFC_OK
+         */ 
+        nfc_err_t cancel_discovery();
+
+    private:
+        friend class NFCRemoteEndpoint;
+        friend class NFCRemoteInitiator;
+        friend class Type4RemoteInitiator;
+        nfc_transceiver_t* transceiver() const;
```

Comment:
This method is :
- private
- non-virtual 
- only returning the private member _transceiver.

`Type4RemoteInitiator` is a friend so it could also directly access the member.
Or making this method public would allow to remove the need for to be a friend.